env/work/srw_python/srwl_smp.py

```diff
@@ -0,0 +1,199 @@
+        for k in self.possible_extensions.keys():
+            for e in self.possible_extensions[k]:
+                if extension == e:
+                    return k
+        all_extensions = [x for x_list in self.possible_extensions.values() for x in x_list]
+        all_extensions += [x.upper() for x in all_extensions]
+        raise ValueError('Incorrect extension: {}. Possible values: {}.'.format(extension, ', '.join(all_extensions)))
+
+
+# ********************** Create transmission element from the data from an image file:
+def srwl_opt_setup_transmission_from_file(file_path, resolution, thickness, delta, atten_len,
+                                          xc=0, yc=0, e_start=0, e_fin=0,
+                                          is_save_images=True, prefix=''):
+    """Setup Sample element.
+
+    :param file_path: path to the input file (.tif or .npy).
+    :param limit_value: maximum possible value (from bits per point).
+    :param nx: number of horizontal points.
+    :param ny: number of vertical points.
+    :param resolution: resolution of the image [m/pixel].
+    :param thickness: thickness of the sample [m].
+    :param delta: refractive index decrement.
+    :param atten_len: attenuation length [m].
+    :param xc: horizontal coordinate of center [m].
+    :param yc: vertical coordinate of center [m].
+    :param e_start: initial photon energy [eV].
+    :param e_fin: final photon energy [eV].
+    :param is_save_images: a flag to save the initial and processed images.
+    :param prefix: the prefix to add to the names of the saved image files.
+    :return: transmission (SRWLOptT) type optical element which simulates the Sample.
+    """
+
+    input_parms = {
+        "type": "sample",
+        "resolution": resolution,
+        "thickness": thickness,
+        "refractiveIndex": delta,
+        "attenuationLength": atten_len,
+        "horizontalCenterCoordinate": xc,
+        "verticalCenterCoordinate": yc,
+        "initialPhotonEnergy": e_start,
+        "finalPhotonPnergy": e_fin,
+    }
+
+    s = SRWLOptSmp(
+        file_path=file_path,
+        is_show_images=False,
+        is_save_images=is_save_images,
+        prefix=prefix,
+    )
+
+    # Input parameters to SRWLOptT:
+    nx = s.nx
+    ny = s.ny
+    rx = nx * resolution
+    ry = ny * resolution
+    fx = 1e+23
+    fy = 1e+23
+    ne = 1
+
+    opT = srwlib.SRWLOptT(nx, ny, rx, ry, None, 1, fx, fy, xc, yc, ne, e_start, e_fin)
+
+    # Same data alignment as for wavefront: outmost loop vs y, inmost loop vs e
+    offset = 0
+    for iy in range(ny):
+        for ix in range(nx):
+            for ie in range(ne):
+                pathInBody = thickness * s.data[ix, iy] / s.limit_value
+                opT.arTr[offset] = math.exp(-0.5 * pathInBody / atten_len)  # amplitude transmission
+                opT.arTr[offset + 1] = -delta * pathInBody  # optical path difference
+                offset += 2
+
+    opT.input_parms = input_parms
+
+    return opT
```

Comment:
1) Make SRWLOptSmp a subclass of SRWLOpt (after a necessary import):
class SRWLOptSmp(SRWLOpt):
2) Wrap 
import numpy as np  
from PIL import Image  
by try-except
Check if SRWLOptSmp class can do something useful without these libraries.